cmd/zfs/zfs_main.c

```diff
@@ -6731,6 +6731,7 @@ zfs_do_holds(int argc, char **argv)
 typedef struct get_all_state {
 	boolean_t	ga_verbose;
 	get_all_cb_t	*ga_cbp;
+	const char	*ga_dataset;
```

Comment:
Who owns this pointer, and do we have to worry about it being deallocated from underneath us?

Reply:
This pointer is set to the value of a command-line argument, so the `exec` system call that launched the process owns it. It will not be deallocated for the life of the program.

Reply:
Thanks @ahesford